### PR TITLE
Update mnnpy: fix deps, cleanup

### PR DESCRIPTION
**Pull request by mbargull**
cc @pinin4fjords
This fixes `mnnpy` by using `skip: True # [py<X]` instead of `python >=X` in the `requirements`.
It is an odd `conda-build` quirk, but we always have to use `skip: True` for all Python packages that are not `noarch: python`.
(We'll update the linter in the future to catch these peculiarities.)

----

Please read the [guidelines for Bioconda recipes](https://bioconda.github.io/contributor/guidelines.html) before opening a pull request (PR).

* If this PR adds or updates a recipe, use "Add" or "Update" appropriately as the first word in its title.
* New recipes not directly relevant to the biological sciences need to be submitted to the [conda-forge channel](https://conda-forge.org/docs/) instead of Bioconda.
* PRs require reviews prior to being merged. Once your PR is passing tests and ready to be merged, please issue the `@BiocondaBot please add label` command.
* Please post questions [on Gitter](https://gitter.im/bioconda/Lobby) or ping `@bioconda/core` in a comment.

<details>
  <summary>Please use the following BiocondaBot commands:</summary>

Everyone has access to the following BiocondaBot commands, which can be given in a comment:

<table>
  <tr>
    <td><code>@BiocondaBot please update</code></td>
    <td>Merge the master branch into a PR.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please add label</code></td>
    <td>Add the <code>please review & merge</code> label.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please fetch artifacts</code></td>
    <td>Post links to CI-built packages/containers. <br />You can use this to test packages locally.</td>
  </tr>
</table>

For members of the Bioconda project, the following command is also available:

<table>
  <tr>
    <td><code>@BiocondaBot please merge</code></td>
    <td>Upload built packages/containers and merge a PR. <br />Someone must approve a PR first! <br />This reduces CI build time by reusing built artifacts.</td>
  </tr>
</table>

Also, the bot watches for comments from non-members that include `@bioconda/<team>` and will automatically re-post them to notify the addressed `<team>`.

</details>

### CODE REVIEW

File: recipes/mnnpy/meta.yaml [hunk 1]
```diff
@@ -1,43 +1,45 @@
...
+  number: 1
+  skip: True  # [py<34]
+  script: {{ PYTHON }} -m pip install . --no-deps --ignore-installed -vv
 
 requirements:
   build:
     - {{ compiler('c') }}
-    - llvm-openmp # [osx]
+    - llvm-openmp  # [osx]
+    - libgomp      # [linux]
```
Comment (mbargull) [inline recipes/mnnpy/meta.yaml:20]: This isn't terribly important, but I'd recommend it as per http://conda-forge.org/docs/maintainer/knowledge_base.html#openmp .

**Comment by mbargull** [comment]
@BiocondaBot, please fetch artifacts

**Comment by mbargull** [comment]
@BiocondaBot, please add label

**Comment by mbargull** [comment]
FYI, the build logs show
```
0:10:53 BIOCONDA INFO (OUT) export SRC_DIR=/opt/conda/conda-bld/mnnpy_1597226704501/test_tmp
10:10:54 BIOCONDA INFO (OUT) /opt/conda/conda-bld/mnnpy_1597226704501/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_pl/lib/python3.7/site-packages/mnnpy/utils.py:14: NumbaWarning:
10:10:54 BIOCONDA INFO (OUT) Compilation is falling back to object mode WITH looplifting enabled because Function "l2_norm" failed type inference due to: No implementation of function Function(<function norm at 0x7f03b4f4bb90>) found for signature:
10:10:55 BIOCONDA INFO (OUT) 
10:10:55 BIOCONDA INFO (OUT)  >>> norm(x=array(float32, 2d, A), axis=Literal[int](1))
10:10:55 BIOCONDA INFO (OUT) 
10:10:55 BIOCONDA INFO (OUT) There are 2 candidate implementations:
10:10:55 BIOCONDA INFO (OUT)   - Of which 2 did not match due to:
10:10:55 BIOCONDA INFO (OUT)   Overload in function 'norm_impl': File: numba/np/linalg.py: Line 2359.
10:10:55 BIOCONDA INFO (OUT)     With argument(s): '(x=array(float32, 2d, A), axis=Literal[int](1))':
10:10:55 BIOCONDA INFO (OUT)    Rejected as the implementation raised a specific error:
10:10:55 BIOCONDA INFO (OUT)      TypeError: norm_impl() got an unexpected keyword argument 'x'
10:10:55 BIOCONDA INFO (OUT)   raised from /opt/conda/conda-bld/mnnpy_1597226704501/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_pl/lib/python3.7/site-packages/numba/core/typing/templates.py:675
10:10:55 BIOCONDA INFO (OUT) 
10:10:55 BIOCONDA INFO (OUT) During: resolving callee type: Function(<function norm at 0x7f03b4f4bb90>)
10:10:55 BIOCONDA INFO (OUT) During: typing of call at /opt/conda/conda-bld/mnnpy_1597226704501/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_pl/lib/python3.7/site-packages/mnnpy/utils.py (16)
10:10:55 BIOCONDA INFO (OUT) 
10:10:55 BIOCONDA INFO (OUT) 
10:10:55 BIOCONDA INFO (OUT) File "../_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_pl/lib/python3.7/site-packages/mnnpy/utils.py", line 16:
10:10:55 BIOCONDA INFO (OUT) def l2_norm(in_matrix):
10:10:55 BIOCONDA INFO (OUT)     return np.linalg.norm(x=in_matrix, axis=1)
10:10:55 BIOCONDA INFO (OUT)     ^
10:10:55 BIOCONDA INFO (OUT) 
10:10:55 BIOCONDA INFO (OUT)   @jit(float32[:](float32[:, :]), nogil=True)
10:10:55 BIOCONDA INFO (OUT) /opt/conda/conda-bld/mnnpy_1597226704501/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_pl/lib/python3.7/site-packages/numba/core/object_mode_passes.py:178: NumbaWarning: Function "l2_norm" was compiled in object mode without forceobj=True.
10:10:55 BIOCONDA INFO (OUT) 
10:10:55 BIOCONDA INFO (OUT) File "../_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_pl/lib/python3.7/site-packages/mnnpy/utils.py", line 15:
10:10:55 BIOCONDA INFO (OUT) @jit(float32[:](float32[:, :]), nogil=True)
10:10:55 BIOCONDA INFO (OUT) def l2_norm(in_matrix):
10:10:55 BIOCONDA INFO (OUT) ^
10:10:55 BIOCONDA INFO (OUT) 
10:10:55 BIOCONDA INFO (OUT)   state.func_ir.loc))
10:10:55 BIOCONDA INFO (OUT) /opt/conda/conda-bld/mnnpy_1597226704501/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_pl/lib/python3.7/site-packages/numba/core/object_mode_passes.py:188: NumbaDeprecationWarning:
10:10:55 BIOCONDA INFO (OUT) Fall-back from the nopython compilation path to the object mode compilation path has been detected, this is deprecated behaviour.
10:10:55 BIOCONDA INFO (OUT) 
10:10:55 BIOCONDA INFO (OUT) For more information visit http://numba.pydata.org/numba-doc/latest/reference/deprecation.html#deprecation-of-object-mode-fall-back-behaviour-when-using-jit
10:10:55 BIOCONDA INFO (OUT) 
10:10:55 BIOCONDA INFO (OUT) File "../_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_pl/lib/python3.7/site-packages/mnnpy/utils.py", line 15:
10:10:55 BIOCONDA INFO (OUT) @jit(float32[:](float32[:, :]), nogil=True)
10:10:55 BIOCONDA INFO (OUT) def l2_norm(in_matrix):
10:10:55 BIOCONDA INFO (OUT) ^
10:10:55 BIOCONDA INFO (OUT) 
10:10:55 BIOCONDA INFO (OUT)   state.func_ir.loc))
10:10:55 BIOCONDA INFO (OUT) /opt/conda/conda-bld/mnnpy_1597226704501/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_pl/lib/python3.7/site-packages/mnnpy/utils.py:14: NumbaWarning: Code running in object mode won't allow parallel execution despite nogil=True.
10:10:55 BIOCONDA INFO (OUT)   @jit(float32[:](float32[:, :]), nogil=True)
10:10:55 BIOCONDA INFO (OUT) /opt/conda/conda-bld/mnnpy_1597226704501/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_pl/lib/python3.7/site-packages/mnnpy/utils.py:29: NumbaPerformanceWarning: np.dot() is faster on contiguous arrays, called on (array(float32, 1d, A), array(float32, 1d, A))
10:10:55 BIOCONDA INFO (OUT)   dist[i, j] = np.dot(m[i], n[j])
10:10:55 BIOCONDA INFO (OUT) /opt/conda/conda-bld/mnnpy_1597226704501/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_pl/lib/python3.7/site-packages/mnnpy/utils.py:88: NumbaWarning:
10:10:55 BIOCONDA INFO (OUT) Compilation is falling back to object mode WITH looplifting enabled because Function "find_mutual_nn" failed type inference due to: Untyped global name 'cKDTree': cannot determine Numba type of <class 'type'>
10:10:55 BIOCONDA INFO (OUT) 
10:10:55 BIOCONDA INFO (OUT) File "../_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_pl/lib/python3.7/site-packages/mnnpy/utils.py", line 90:
10:10:55 BIOCONDA INFO (OUT) def find_mutual_nn(data1, data2, k1, k2, n_jobs):
10:10:55 BIOCONDA INFO (OUT)     k_index_1 = cKDTree(data1).query(x=data2, k=k1, n_jobs=n_jobs)[1]
10:10:55 BIOCONDA INFO (OUT)     ^
10:10:55 BIOCONDA INFO (OUT) 
10:10:55 BIOCONDA INFO (OUT)   @jit((float32[:, :], float32[:, :], int8, int8, int8))
10:10:55 BIOCONDA INFO (OUT) /opt/conda/conda-bld/mnnpy_1597226704501/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_pl/lib/python3.7/site-packages/mnnpy/utils.py:88: NumbaWarning:
10:10:55 BIOCONDA INFO (OUT) Compilation is falling back to object mode WITHOUT looplifting enabled because Function "find_mutual_nn" failed type inference due to: Untyped global name 'cKDTree': cannot determine Numba type of <class 'type'>
10:10:55 BIOCONDA INFO (OUT) 
10:10:55 BIOCONDA INFO (OUT) File "../_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_pl/lib/python3.7/site-packages/mnnpy/utils.py", line 90:
10:10:55 BIOCONDA INFO (OUT) def find_mutual_nn(data1, data2, k1, k2, n_jobs):
10:10:55 BIOCONDA INFO (OUT)     k_index_1 = cKDTree(data1).query(x=data2, k=k1, n_jobs=n_jobs)[1]
10:10:55 BIOCONDA INFO (OUT)     ^
10:10:55 BIOCONDA INFO (OUT) 
10:10:55 BIOCONDA INFO (OUT)   @jit((float32[:, :], float32[:, :], int8, int8, int8))
10:10:55 BIOCONDA INFO (OUT) /opt/conda/conda-bld/mnnpy_1597226704501/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_pl/lib/python3.7/site-packages/numba/core/object_mode_passes.py:178: NumbaWarning: Function "find_mutual_nn" was compiled in object mode without forceobj=True, but has lifted loops.
10:10:55 BIOCONDA INFO (OUT) 
10:10:55 BIOCONDA INFO (OUT) File "../_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_pl/lib/python3.7/site-packages/mnnpy/utils.py", line 90:
10:10:55 BIOCONDA INFO (OUT) def find_mutual_nn(data1, data2, k1, k2, n_jobs):
10:10:55 BIOCONDA INFO (OUT)     k_index_1 = cKDTree(data1).query(x=data2, k=k1, n_jobs=n_jobs)[1]
10:10:55 BIOCONDA INFO (OUT)     ^
10:10:55 BIOCONDA INFO (OUT) 
10:10:55 BIOCONDA INFO (OUT)   state.func_ir.loc))
10:10:55 BIOCONDA INFO (OUT) /opt/conda/conda-bld/mnnpy_1597226704501/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_pl/lib/python3.7/site-packages/numba/core/object_mode_passes.py:188: NumbaDeprecationWarning:
10:10:55 BIOCONDA INFO (OUT) Fall-back from the nopython compilation path to the object mode compilation path has been detected, this is deprecated behaviour.
10:10:55 BIOCONDA INFO (OUT) 
10:10:55 BIOCONDA INFO (OUT) For more information visit http://numba.pydata.org/numba-doc/latest/reference/deprecation.html#deprecation-of-object-mode-fall-back-behaviour-when-using-jit
10:10:55 BIOCONDA INFO (OUT) 
10:10:55 BIOCONDA INFO (OUT) File "../_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_pl/lib/python3.7/site-packages/mnnpy/utils.py", line 90:
10:10:55 BIOCONDA INFO (OUT) def find_mutual_nn(data1, data2, k1, k2, n_jobs):
10:10:55 BIOCONDA INFO (OUT)     k_index_1 = cKDTree(data1).query(x=data2, k=k1, n_jobs=n_jobs)[1]
10:10:55 BIOCONDA INFO (OUT)     ^
10:10:55 BIOCONDA INFO (OUT) 
10:10:55 BIOCONDA INFO (OUT)   state.func_ir.loc))
10:10:55 BIOCONDA INFO (OUT) /opt/conda/conda-bld/mnnpy_1597226704501/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_pl/lib/python3.7/site-packages/mnnpy/utils.py:102: NumbaWarning:
10:10:55 BIOCONDA INFO (OUT) Compilation is falling back to object mode WITH looplifting enabled because Function "compute_correction" failed type inference due to: No implementation of function Function(<function unique at 0x7f03b4e960e0>) found for signature:
10:10:55 BIOCONDA INFO (OUT) 
10:10:55 BIOCONDA INFO (OUT)  >>> unique(array(int32, 1d, A), return_counts=bool)
10:10:55 BIOCONDA INFO (OUT) 
10:10:55 BIOCONDA INFO (OUT) There are 2 candidate implementations:
10:10:55 BIOCONDA INFO (OUT)   - Of which 2 did not match due to:
10:10:55 BIOCONDA INFO (OUT)   Overload in function 'np_unique': File: numba/np/arrayobj.py: Line 1915.
10:10:55 BIOCONDA INFO (OUT)     With argument(s): '(array(int32, 1d, A), return_counts=bool)':
10:10:55 BIOCONDA INFO (OUT)    Rejected as the implementation raised a specific error:
10:10:55 BIOCONDA INFO (OUT)      TypeError: np_unique() got an unexpected keyword argument 'return_counts'
10:10:55 BIOCONDA INFO (OUT)   raised from /opt/conda/conda-bld/mnnpy_1597226704501/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_pl/lib/python3.7/site-packages/numba/core/typing/templates.py:675
10:10:55 BIOCONDA INFO (OUT) 
10:10:55 BIOCONDA INFO (OUT) During: resolving callee type: Function(<function unique at 0x7f03b4e960e0>)
10:10:55 BIOCONDA INFO (OUT) During: typing of call at /opt/conda/conda-bld/mnnpy_1597226704501/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_pl/lib/python3.7/site-packages/mnnpy/utils.py (105)
10:10:55 BIOCONDA INFO (OUT) 
10:10:55 BIOCONDA INFO (OUT) 
10:10:55 BIOCONDA INFO (OUT) File "../_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_pl/lib/python3.7/site-packages/mnnpy/utils.py", line 105:
10:10:55 BIOCONDA INFO (OUT) def compute_correction(data1, data2, mnn1, mnn2, data2_or_raw2, sigma):
10:10:55 BIOCONDA INFO (OUT)     <source elided>
10:10:55 BIOCONDA INFO (OUT)     vect = data1[mnn1] - data2[mnn2]
10:10:55 BIOCONDA INFO (OUT)     mnn_index, mnn_count = np.unique(mnn2, return_counts=True)
10:10:55 BIOCONDA INFO (OUT)     ^
10:10:55 BIOCONDA INFO (OUT) 
10:10:55 BIOCONDA INFO (OUT)   @jit(float32[:, :](float32[:, :], float32[:, :], int32[:], int32[:], float32[:, :], float32))
10:10:55 BIOCONDA INFO (OUT) /opt/conda/conda-bld/mnnpy_1597226704501/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_pl/lib/python3.7/site-packages/mnnpy/utils.py:102: NumbaWarning:
10:10:55 BIOCONDA INFO (OUT) Compilation is falling back to object mode WITHOUT looplifting enabled because Function "compute_correction" failed type inference due to: cannot determine Numba type of <class 'numba.core.dispatcher.LiftedLoop'>
10:10:55 BIOCONDA INFO (OUT) 
10:10:55 BIOCONDA INFO (OUT) File "../_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_pl/lib/python3.7/site-packages/mnnpy/utils.py", line 107:
10:10:55 BIOCONDA INFO (OUT) def compute_correction(data1, data2, mnn1, mnn2, data2_or_raw2, sigma):
10:10:55 BIOCONDA INFO (OUT)     <source elided>
10:10:55 BIOCONDA INFO (OUT)     vect_reduced = np.zeros((data2.shape[0], vect.shape[1]), dtype=np.float32)
10:10:55 BIOCONDA INFO (OUT)     for index, ve in zip(mnn2, vect):
10:10:55 BIOCONDA INFO (OUT)     ^
10:10:55 BIOCONDA INFO (OUT) 
10:10:55 BIOCONDA INFO (OUT)   @jit(float32[:, :](float32[:, :], float32[:, :], int32[:], int32[:], float32[:, :], float32))
10:10:56 BIOCONDA INFO (OUT) /opt/conda/conda-bld/mnnpy_1597226704501/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_pl/lib/python3.7/site-packages/numba/core/object_mode_passes.py:178: NumbaWarning: Function "compute_correction" was compiled in object mode without forceobj=True, but has lifted loops.
10:10:56 BIOCONDA INFO (OUT) 
10:10:56 BIOCONDA INFO (OUT) File "../_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_pl/lib/python3.7/site-packages/mnnpy/utils.py", line 104:
10:10:56 BIOCONDA INFO (OUT) def compute_correction(data1, data2, mnn1, mnn2, data2_or_raw2, sigma):
10:10:56 BIOCONDA INFO (OUT)     vect = data1[mnn1] - data2[mnn2]
10:10:56 BIOCONDA INFO (OUT)     ^
10:10:56 BIOCONDA INFO (OUT) 
10:10:56 BIOCONDA INFO (OUT)   state.func_ir.loc))
10:10:56 BIOCONDA INFO (OUT) /opt/conda/conda-bld/mnnpy_1597226704501/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_pl/lib/python3.7/site-packages/numba/core/object_mode_passes.py:188: NumbaDeprecationWarning:
10:10:56 BIOCONDA INFO (OUT) Fall-back from the nopython compilation path to the object mode compilation path has been detected, this is deprecated behaviour.
10:10:56 BIOCONDA INFO (OUT) 
10:10:56 BIOCONDA INFO (OUT) For more information visit http://numba.pydata.org/numba-doc/latest/reference/deprecation.html#deprecation-of-object-mode-fall-back-behaviour-when-using-jit
10:10:56 BIOCONDA INFO (OUT) 
10:10:56 BIOCONDA INFO (OUT) File "../_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_pl/lib/python3.7/site-packages/mnnpy/utils.py", line 104:
10:10:56 BIOCONDA INFO (OUT) def compute_correction(data1, data2, mnn1, mnn2, data2_or_raw2, sigma):
10:10:56 BIOCONDA INFO (OUT)     vect = data1[mnn1] - data2[mnn2]
10:10:56 BIOCONDA INFO (OUT)     ^
10:10:56 BIOCONDA INFO (OUT) 
10:10:56 BIOCONDA INFO (OUT)   state.func_ir.loc))
10:10:56 BIOCONDA INFO (OUT) /opt/conda/conda-bld/mnnpy_1597226704501/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_pl/lib/python3.7/site-packages/mnnpy/utils.py:199: NumbaWarning:
10:10:56 BIOCONDA INFO (OUT) Compilation is falling back to object mode WITH looplifting enabled because Function "adjust_s_variance" failed type inference due to: NameError: name 'sq_dist_to_line' is not defined
10:10:56 BIOCONDA INFO (OUT)   @jit(float32(float32[:, :], float32[:, :], float32[:], float32[:], float32), nogil=True)
10:10:56 BIOCONDA INFO (OUT) /opt/conda/conda-bld/mnnpy_1597226704501/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_pl/lib/python3.7/site-packages/mnnpy/utils.py:199: NumbaWarning:
10:10:56 BIOCONDA INFO (OUT) Compilation is falling back to object mode WITHOUT looplifting enabled because Function "adjust_s_variance" failed type inference due to: cannot determine Numba type of <class 'numba.core.dispatcher.LiftedLoop'>
10:10:56 BIOCONDA INFO (OUT) 
10:10:56 BIOCONDA INFO (OUT) File "../_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_pl/lib/python3.7/site-packages/mnnpy/utils.py", line 207:
10:10:56 BIOCONDA INFO (OUT) def adjust_s_variance(data1, data2, curcell, curvect, sigma):
10:10:56 BIOCONDA INFO (OUT)     <source elided>
10:10:56 BIOCONDA INFO (OUT)     totalprob2 = 0.
10:10:56 BIOCONDA INFO (OUT)     for samecell in data2:
10:10:56 BIOCONDA INFO (OUT)     ^
10:10:56 BIOCONDA INFO (OUT) 
10:10:56 BIOCONDA INFO (OUT)   @jit(float32(float32[:, :], float32[:, :], float32[:], float32[:], float32), nogil=True)
10:10:56 BIOCONDA INFO (OUT) /opt/conda/conda-bld/mnnpy_1597226704501/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_pl/lib/python3.7/site-packages/numba/core/object_mode_passes.py:178: NumbaWarning: Function "adjust_s_variance" was compiled in object mode without forceobj=True, but has lifted loops.
10:10:56 BIOCONDA INFO (OUT) 
10:10:56 BIOCONDA INFO (OUT) File "../_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_pl/lib/python3.7/site-packages/mnnpy/utils.py", line 201:
10:10:56 BIOCONDA INFO (OUT) def adjust_s_variance(data1, data2, curcell, curvect, sigma):
10:10:56 BIOCONDA INFO (OUT)     distance1 = np.zeros((data1.shape[0], 2), dtype=np.float32)
10:10:56 BIOCONDA INFO (OUT)     ^
10:10:56 BIOCONDA INFO (OUT) 
10:10:56 BIOCONDA INFO (OUT)   state.func_ir.loc))
10:10:56 BIOCONDA INFO (OUT) /opt/conda/conda-bld/mnnpy_1597226704501/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_pl/lib/python3.7/site-packages/numba/core/object_mode_passes.py:188: NumbaDeprecationWarning:
10:10:56 BIOCONDA INFO (OUT) Fall-back from the nopython compilation path to the object mode compilation path has been detected, this is deprecated behaviour.
10:10:56 BIOCONDA INFO (OUT) 
10:10:56 BIOCONDA INFO (OUT) For more information visit http://numba.pydata.org/numba-doc/latest/reference/deprecation.html#deprecation-of-object-mode-fall-back-behaviour-when-using-jit
10:10:56 BIOCONDA INFO (OUT) 
10:10:56 BIOCONDA INFO (OUT) File "../_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_pl/lib/python3.7/site-packages/mnnpy/utils.py", line 201:
10:10:56 BIOCONDA INFO (OUT) def adjust_s_variance(data1, data2, curcell, curvect, sigma):
10:10:56 BIOCONDA INFO (OUT)     distance1 = np.zeros((data1.shape[0], 2), dtype=np.float32)
10:10:56 BIOCONDA INFO (OUT)     ^
10:10:56 BIOCONDA INFO (OUT) 
10:10:56 BIOCONDA INFO (OUT)   state.func_ir.loc))
10:10:56 BIOCONDA INFO (OUT) /opt/conda/conda-bld/mnnpy_1597226704501/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_pl/lib/python3.7/site-packages/mnnpy/utils.py:199: NumbaWarning: Code running in object mode won't allow parallel execution despite nogil=True.
10:10:56 BIOCONDA INFO (OUT)   @jit(float32(float32[:, :], float32[:, :], float32[:], float32[:], float32), nogil=True)
10:10:56 BIOCONDA INFO (OUT) /opt/conda/conda-bld/mnnpy_1597226704501/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_pl/lib/python3.7/site-packages/mnnpy/utils.py:238: NumbaPerformanceWarning: np.dot() is faster on contiguous arrays, called on (array(float32, 1d, C), array(float32, 1d, A))
10:10:56 BIOCONDA INFO (OUT)   scale = np.dot(working, grad)
10:10:57 BIOCONDA INFO (OUT) import: 'mnnpy'
```

**Comment by pinin4fjords** [comment]
Hmm, looking at a number of issues on the source repo https://github.com/chriscainx/mnnpy/issues?q=numba, I think pinning Numba might help- I'll give it a go.

**Comment by mbargull** [comment]
Seems like it's only one file that uses `numba.jit`. We might be able to simply patch it, too. I'll have a look.

**Comment by pinin4fjords** [comment]
@mbargull you're a wizard!

**Comment by mbargull** [comment]
Cool, worked, that was easy, thankfully :).
>  Good to merge?

Yes, let me have a quick look at the artifacts. When you approve this PR, I can instruct the bot to merge ;).
> Or should we PR the package?

I don't follow. Do you mean whether we should upstream the patch? If you could do that at https://github.com/chriscainx/mnnpy/pulls, I'd very much appreciate that!
(The upstream project doesn't seem to be maintained currently, hence I think it makes sense that we include the change as a patch here for now.)

**Comment by mbargull** [comment]
@BiocondaBot, please fetch artifacts

**Comment by pinin4fjords** [comment]
> > Or should we PR the package?
> 
> I don't follow. Do you mean whether we should upstream the patch? If you could do that at https://github.com/chriscainx/mnnpy/pulls, I'd very much appreciate that!

Yes, that's what I meant. Happy to oblige.

**Comment by mbargull** [comment]
> Yes, that's what I meant. Happy to oblige.

Thank you!

**Comment by mbargull** [comment]
@BiocondaBot, please merge

**Pull request merged**